diff --git a/nose2/plugins/junitxml.py b/nose2/plugins/junitxml.py
index 32dd214..cbf2813 100644
--- a/nose2/plugins/junitxml.py
+++ b/nose2/plugins/junitxml.py
@@ -39,16 +39,16 @@ Produces this XML by default:

 .. code-block:: xml

-    <testcase classname="a" name="test_gen:1" time="0.000171">
+    <testcase classname="a" name="test_gen:1" time="0.000171" timestamp="2021-12-09T21:28:09.686611">
         <system-out />
     </testcase>
-    <testcase classname="a" name="test_gen:2" time="0.000202">
+    <testcase classname="a" name="test_gen:2" time="0.000202" timestamp="2021-12-09T21:28:09.686813">
         <system-out />
     </testcase>
-    <testcase classname="a" name="test_params:1" time="0.000159">
+    <testcase classname="a" name="test_params:1" time="0.000159" timestamp="2021-12-09T21:28:09.686972">
         <system-out />
     </testcase>
-    <testcase classname="a" name="test_params:2" time="0.000163">
+    <testcase classname="a" name="test_params:2" time="0.000163" timestamp="2021-12-09T21:28:09.687135">
         <system-out />
     </testcase>

@@ -57,16 +57,16 @@ produced:

 .. code-block:: xml

-    <testcase classname="a" name="test_gen:1 (99, 99)" time="0.000213">
+    <testcase classname="a" name="test_gen:1 (99, 99)" time="0.000213" timestamp="2021-12-09T21:28:09.686611">
         <system-out />
     </testcase>
-    <testcase classname="a" name="test_gen:2 (-1, -1)" time="0.000194">
+    <testcase classname="a" name="test_gen:2 (-1, -1)" time="0.000194" timestamp="2021-12-09T21:28:09.687105">
         <system-out />
     </testcase>
-    <testcase classname="a" name="test_params:1 ('foo')" time="0.000178">
+    <testcase classname="a" name="test_params:1 ('foo')" time="0.000178" timestamp="2021-12-09T21:28:09.687283">
         <system-out />
     </testcase>
-    <testcase classname="a" name="test_params:2 ('bar')" time="0.000187">
+    <testcase classname="a" name="test_params:2 ('bar')" time="0.000187" timestamp="2021-12-09T21:28:09.687470">
         <system-out />
     </testcase>

@@ -79,6 +79,7 @@ import os.path
 import re
 import sys
 import time
+import datetime
 from xml.etree import ElementTree as ET

 import six
@@ -152,6 +153,8 @@ class JUnitXmlReporter(events.Plugin):
             return

         testcase = ET.SubElement(self.tree, "testcase")
+        testcase.set('timestamp',
+                     str(self._timestamp_to_iso8601(self._start)))
         testcase.set("time", "%.6f" % self._time())
         if not classname:
             classname = test.__module__
@@ -285,6 +288,8 @@ class JUnitXmlReporter(events.Plugin):
             self._start = None
         return 0

+    def _timestamp_to_iso8601(self, timestamp):
+        return datetime.datetime.utcfromtimestamp(timestamp).isoformat()

 #
 # xml utility functions
